python/huggingfaceserver/huggingfaceserver/encoder_model.py

```diff
@@ -0,0 +1,279 @@
+        tensor_input_names: Optional[str] = None,
+        return_token_type_ids: Optional[bool] = None,
+        model_revision: Optional[str] = None,
+        tokenizer_revision: Optional[str] = None,
+        trust_remote_code: bool = False,
+        predictor_config: Optional[PredictorConfig] = None,
+    ):
+        super().__init__(model_name, predictor_config)
+        self._device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+        self.model_id_or_path = model_id_or_path
+        self.do_lower_case = do_lower_case
+        self.add_special_tokens = add_special_tokens
+        self.max_length = max_length
+        self.tensor_input_names = tensor_input_names
+        self.return_token_type_ids = return_token_type_ids
+        self.model_revision = model_revision
+        self.tokenizer_revision = tokenizer_revision
+        self.trust_remote_code = trust_remote_code
+
+        if model_config:
+            self.model_config = model_config
+        else:
+            self.model_config = AutoConfig.from_pretrained(self.model_id_or_path)
+
+        if task:
+            self.task = task
+            try:
+                inferred_task = infer_task_from_model_architecture(self.model_config)
+            except ValueError:
+                inferred_task = None
+            if inferred_task is not None and inferred_task != task:
+                logger.warn(
+                    f"Inferred task is '{inferred_task.name}' but"
+                    f" task is explicitly set to '{self.task.name}'"
+                )
+        else:
+            self.task = infer_task_from_model_architecture(self.model_config)
+
+        if is_generative_task(self.task):
+            raise RuntimeError(
+                f"Encoder model does not support generative task: {self.task.name}"
+            )
+
+    def load(self) -> bool:
+        model_id_or_path = self.model_id_or_path
+        if self.max_length is None:
+            self.max_length = self.model_config.max_length
+
+        # device_map = "auto" enables model parallelism but all model architcture dont support it.
+        # For pre-check we initialize the model class without weights to check the `_no_split_modules`
+        # device_map = "auto" for models that support this else set to either cuda/cpu
+        with init_empty_weights():
+            self._model = AutoModel.from_config(self.model_config)
+
+        device_map = self._device
+
+        if self._model._no_split_modules:
+            device_map = "auto"
+
+        tokenizer_kwargs = {}
+        model_kwargs = {}
+
+        if self.trust_remote_code:
+            model_kwargs["trust_remote_code"] = True
+            tokenizer_kwargs["trust_remote_code"] = True
+
+        # load huggingface tokenizer
+        self._tokenizer = AutoTokenizer.from_pretrained(
+            str(model_id_or_path),
+            revision=self.tokenizer_revision,
+            do_lower_case=self.do_lower_case,
```

Comment:
Can you once verify the tokenizer args? Tokenizer also has device_map setting.

Reply:
I am having a hard time finding any reference to `device_map` in the HF transformers code. There's also no mention of tokenizers supporting `device_map` [in the docs](https://huggingface.co/docs/accelerate/concept_guides/big_model_inference#the-devicemap).  This comment suggests it's not needed/supported: https://github.com/huggingface/transformers/issues/16359#issuecomment-2075703924

Reply:
Let me try this again tomorrow. Btw, model used was gemma-2b

Reply:
I also can not find anything from [gemma-2b](https://huggingface.co/google/gemma-2b-it)